src/components/FormInput.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+class FormInput extends React.Component {
+
+  constructor() {
+    super();
+    this.state = { name:'', errorMessage:'' }
+    this.onInputChange = this.onInputChange.bind(this);
+  }   
+
+  onInputChange(event) {
+    let type = event.target.type;
+    let value = event.target.value;
+    this.validateField(type, value);
+  }
+    
+  validateField(type, value) {
+
+    let errorMessage = this.state.errorMessage;
```

Comment:
```suggestion
    let errorMessage = '';
```
this is fine, you are not using the `this.state.errorMessage` down the line. By doing this you dont have to reset the `errorMessage` in lines 26-28 (remove the else condition)